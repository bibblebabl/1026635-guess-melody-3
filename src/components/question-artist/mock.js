export const question = {
  type: `artist`,
  question: `Вопрос?`,
  answers: [
    {
      id: 1,
      label: `Ответ-1`,
      value: `Ответ-1`,
      url: `http://placehold.it/134x134`
    },
    {
      id: 2,
      label: `Ответ-2`,
      value: `Ответ-2`,
      url: `http://placehold.it/134x134`
    },
    {
      id: 446,
      label: `Ответ-3`,
      value: `Ответ-3`,
      url: `http://placehold.it/134x134`
    },
  ]
};
