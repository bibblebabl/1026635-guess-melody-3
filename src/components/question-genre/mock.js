export const question = {
  type: `genre`,
  question: `Вопрос`,
  answers: [
    {
      id: 1,
      value: `трек-1`,
      url: `http://dssdsd.ru`
    },
    {
      id: 2,
      value: `трек-2`,
      url: `http://dssdsd.ru`
    },
    {
      id: 3,
      value: `трек-3`,
      url: `http://dssdsd.ru`
    },
    {
      id: 4,
      value: `трек-4`,
      url: `http://dssdsd.ru`
    },
  ]
};
