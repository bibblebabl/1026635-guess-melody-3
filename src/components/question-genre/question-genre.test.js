import React from 'react';
import renderer from 'react-test-renderer';
import QuestionGenre from './question-genre';

import {question} from './mock';

it(`<QuestionGenre /> renders correctly`, () => {
  const component = renderer.create(
      <QuestionGenre onAnswer={() => {}} question={question} />
  ).toJSON();

  expect(component).toMatchSnapshot();
});
