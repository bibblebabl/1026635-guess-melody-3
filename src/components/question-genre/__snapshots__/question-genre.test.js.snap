// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionGenre /> renders correctly 1`] = `
<section
  className="game game--genre"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx={390}
        cy={390}
        r={370}
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Вопрос
    </h2>
    <form
      className="game__tracks"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            defaultValue="трек-1"
            id={1}
            name="Вопрос"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="game__check"
            htmlFor={1}
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            defaultValue="трек-2"
            id={2}
            name="Вопрос"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="game__check"
            htmlFor={2}
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            defaultValue="трек-3"
            id={3}
            name="Вопрос"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="game__check"
            htmlFor={3}
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            defaultValue="трек-4"
            id={4}
            name="Вопрос"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="game__check"
            htmlFor={4}
          >
            Отметить
          </label>
        </div>
      </div>
      <button
        className="game__submit button"
        type="submit"
      >
        Ответить
      </button>
    </form>
  </section>
</section>
`;
